creator/ingest_runs/common/model.py

```diff
@@ -0,0 +1,182 @@
+import hashlib
+import uuid
+import re
+
+import django_rq
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ValidationError
+from django.db import models
+from django_fsm import FSMField, transition
+
+from creator.jobs.models import JobLog
+from creator.utils import stop_job
+
+User = get_user_model()
+
+
+def camel_to_snake(camel_str):
+    """
+    Convert CamelCase to snake_case
+    """
+    return re.sub(r'(?<!^)(?=[A-Z])', '_', camel_str).lower()
+
+
+def hash_versions(versions):
```

Comment:
This requires a list of versions to be passed which may often require a bunch of unneeded fields (you only need the kf_id of each version).
You could consider accepting a list of kf_ids instead so you could use `values()` or `values_list()` to fetch only the kf_id field without having to pull the complete version row.